orcid_service/views.py

```diff
@@ -94,21 +100,34 @@ def orcid_profile_local(orcid_id, type):
     if type not in ['simple','full']:
         return json.dumps('Endpoint /orcid-profile/%s does not exist'.format(type)), 404
 
-    r = current_app.client.get(current_app.config['ORCID_API_ENDPOINT'] + '/' + orcid_id + '/record',
+    try:
+        r = current_app.client.get(current_app.config['ORCID_API_ENDPOINT'] + '/' + orcid_id + '/record',
                                    headers=headers)
+    except requests.exceptions.ConnectionError:
+        logging.error('For ORCID ID {0}, there was a connection error with the ORCID API'.format(orcid_id))
+        return 'There was a connection error with the ORCID API', 502
 
     if r.status_code == 200:
         update_profile_local(orcid_id, data=r.text, force=update)
     else:
-        logging.warning('Failed fetching fresh profile from ORCID for %s'.format(orcid_id))
+        logging.warning('Failed fetching fresh profile from ORCID for {0}. Status {1}, error: {2}'.format(orcid_id,
```

Comment:
It's not very important in this case, but so that you know, if you do:

```
logging.warning('Failed %s', variable)
```

The whole thing will only run if the logging level includes the warnings (the formatting gets executed only if the warning is going to be registered). While if you have:


```
logging.warning('Failed {}'.format(variable))
```

The part `'Failed {}'.format(variable)` will get executed even if the logging level does not include the warning level. So it's best practices to follow the first pattern, although I also forget sometimes!